tests/unit/h/services/group_list_test.py

```diff
@@ -161,7 +158,8 @@ class TestUserGroups:
     def test_it_returns_all_user_groups_sorted_by_group_name(
         self, svc, user, user_groups
     ):
-        user.groups = user_groups
+        user.memberships = []
+        user.memberships.extend([GroupMembership(group=group) for group in user_groups])
```

Comment:
Here I need to actually remove existing memberships in `user.memberships` as well or the test fails, so it's not enough to just call `user.memberships.extend([...])`. If I do `user.memberships = [...]` I'm running into that SQLAlchemy warning again:

```
sqlalchemy.exc.SAWarning: Object of type <GroupMembership> not in session, add operation along 'Group.memberships' will not proceed (This warning originated from the Session 'autoflush' process, which was invoked automatically in response to a user-initiated operation.)
```

So I'm having to first do `user.memberships = []` to remove the existing memberships and then call `extend()`